File: Python/RegExp.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

import re

print("\r\nこれから正規表現を試します\r\n")

text="The Quick Brown Fox Jumps Over The Lazy Dog"
print(f"{text} → ")

print("\r\n検索")
found=re.search(r"(?i)the ([a-z]+) ([a-z]+) fox",text).group(0)
print("re.search: "+found)
'''
foundには,正規表現がマッチした際にはマッチオブジェクトが格納される。
マッチオブジェクトはテキスト中で最初にマッチした部分しか取り扱わない。
found.group(n) : n番目の結果が返される
	n=0  : マッチしたテキスト全体
	n=1… : 正規表現中のn番目の括弧にマッチしたテキスト
なお,マッチしなかった場合は,foundにはNoneが代入されている。
'''

print("\r\n置換")
# 正規表現が使えない置換
replaced=text.replace("Jump","Skip")
print("str.replace:\r\n"+replaced)
# 正規表現が使える置換
replaced=re.sub(r"(?i)([a-z]+)o([a-z]+)",r"« \1ö\2 »",text)
print("re.sub:\r\n"+replaced)

print("\r\n分割と結合")
array=text.split(r"(?i)(the|s)? ")
arranged="_".join(array)
print(arranged)

print("\r\n")

File: JavaScript/RegExp.js
#! /usr/bin/env node

var log="\r\nこれから正規表現を試します\r\n\r\n";

let text="The Quick Brown Fox Jumps Over The Lazy Dog";
log+=`${text} →`;

log+="\r\n検索\r\n";
var found=text.match(/the ([a-z]+) ([a-z]+) fox/g);
log+=`String.match:\r\n${found}\r\n`;

log+="\r\n置換\r\n";
var replaced=text.replace(/([a-z]+)o([a-z]+)/i,"« $1ö$2 »");
log+=`gフラグなし:\r\n${replaced}\r\n`;
var replaced=text.replace(/([a-z]+)o([a-z]+)/ig,"« $1ö$2 »");
log+=`gフラグあり:\r\n${replaced}\r\n`;

log+="\r\n分割と結合\r\n";
let array=text.split(/(the|s)? /);
arranged=array.join("_");
log+=`${array}\r\n`;
log+=`${arranged}\r\n`;

log+="\r\nマッチの確認\r\n";
if (/fox jumps/i.test(text)) log+=`狐が飛んでいます\r\n`;
else log+=`狐は飛んでいません\r\n`;
if (/dog jumps/i.test(text)) log+=`犬が飛んでいます\r\n`;
else log+=`犬は飛んでいません\r\n`;

console.log(log);